pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -305,76 +305,64 @@ func (a Adapter) waitAndGetComponentPod(hideSpinner bool) (*corev1.Pod, error) {
+		// Get Init Command
+		command, ok := commandsMap[versionsCommon.InitCommandGroupType]
+		if ok {
+			compInfo.ContainerName = command.Exec.Component
+			err = exec.ExecuteDevfileBuildAction(&a.Client, *command.Exec, command.Exec.Id, compInfo, show)
+			if err != nil {
+				return err
+			}
+
+		}
+
 	}
-	commandOrder = append(
-		commandOrder,
-		common.CommandNames{DefaultName: string(common.DefaultDevfileBuildCommand), AdapterName: a.devfileBuildCmd},
-		common.CommandNames{DefaultName: string(common.DefaultDevfileRunCommand), AdapterName: a.devfileRunCmd},
-	)
-
-	// Loop through each of the expected commands in the devfile
-	for i, currentCommand := range commandOrder {
-		// Loop through each of the command given from the devfile
-		for _, command := range pushDevfileCommands {
-			// If the current command from the devfile is the currently expected command from the devfile
-			if command.Exec.Id == currentCommand.DefaultName || command.Exec.Id == currentCommand.AdapterName {
-				// If the current command is not the last command in the slice
-				// it is not expected to be the run command
-				if i < len(commandOrder)-1 {
-					// Any exec command such as "Init" and "Build"
-
-					compInfo := common.ComponentInfo{
-						ContainerName: command.Exec.Component,
-						PodName:       podName,
-					}
-
-					err = exec.ExecuteDevfileBuildAction(&a.Client, *command.Exec, command.Exec.Id, compInfo, show)
-					if err != nil {
-						return err
-					}
-
-					// If the current command is the last command in the slice
-					// it is expected to be the run command
-				} else {
-					// Last command is "Run"
-					klog.V(4).Infof("Executing devfile command %v", command.Exec.Id)
-
-					// Check if the devfile run component containers have supervisord as the entrypoint.
-					// Start the supervisord if the odo component does not exist
-					if !componentExists {
-						err = a.InitRunContainerSupervisord(command.Exec.Component, podName, containers)
-						if err != nil {
-							return
-						}
-					}
-
-					compInfo := common.ComponentInfo{
-						ContainerName: command.Exec.Component,
-						PodName:       podName,
-					}
-
-					if componentExists && !common.IsRestartRequired(command) {
-						klog.V(4).Infof("restart:false, Not restarting DevRun Command")
-						err = exec.ExecuteDevfileRunActionWithoutRestart(&a.Client, *command.Exec, command.Exec.Id, compInfo, show)
-						return
-					}
-
-					err = exec.ExecuteDevfileRunAction(&a.Client, *command.Exec, command.Exec.Id, compInfo, show)
+
+	// Get Build Command
+	command, ok := commandsMap[versionsCommon.BuildCommandGroupType]
+	if ok {
+		compInfo.ContainerName = command.Exec.Component
+		err = exec.ExecuteDevfileBuildAction(&a.Client, *command.Exec, command.Exec.Id, compInfo, show)
+		if err != nil {
+			return err
+		}
+
+		command, ok := commandsMap[versionsCommon.RunCommandGroupType]
```

Comment:
why is the run command run only if a build command is present?

Reply:
my mistake, updated